gulp/sass.js

```diff
@@ -5,15 +5,32 @@
   also includes sourcemaps
 */
 
+const fs = require('fs')
 const gulp = require('gulp')
+const path = require('path')
 const sass = require('gulp-sass')
+const sassVariables = require('gulp-sass-variables')
 const sourcemaps = require('gulp-sourcemaps')
 
 const config = require('./config.json')
 
+// Default cache prefix
+let cacheId = ''
+
+// Inject Sass variables
+const variables = () => {
+  cacheId = cacheId || fs.readFileSync(path.resolve('./app/version.txt'), 'utf-8').trim()
```

Comment:
I think it'd be more robust to fingerprint the assets themselves?

For example https://www.npmjs.com/package/gulp-fingerprint

Reply:
Yup this would be the best approach! Significantly more work 😊

E.g. we'd need to update `govuk-font-url()` and `govuk-image-url()` to add a hash/version path: https://github.com/alphagov/govuk-frontend/blob/c809239e05c34c5adb578e7420090478a169a1e5/src/tools/_font-url.scss

Otherwise the URL will be cached for a year without cache-busting.

Reply:
@nickcolley Given this some more thought.

To use per-file fingerprinting (e.g. `gulp-rev`) by adding hashes to all included filenames, we'd need to dynamically patch:

```
app/views/includes/head.html
app/views/includes/scripts.html
```

Since the prototype kit doesn't follow the usual `app`/`src` ==> `dist` "copy on build" step that others do this would be quite tricky to do without big build changes and watch tasks.

The `git-rev` gulp plugin _can_ write to a manifest file (to track files renamed with added hashes) but then the running Node.js application would have to be aware of this to include the correct CSS/JS file.

Have you had any other thoughts?

Reply:
I believe we'd need to use a helper function in the Nunjucks view that references the manifest generated by `gulp-rev`?

Reply:
Does that not feel too tied in to specific tooling? It's what I started doing, wanted to check first. Leave it with me then